### ISSUE TITLE
Bugs: About proba or not proba

### ISSUE THREAD

**Issue by HugoPhi** [HugoPhi/lrkit#1]
![Image](https://github.com/user-attachments/assets/e16e8be1-20e2-4979-8cb9-14255d1db906)

I past in y: proba, y_pred: not proba. The code in Metric should be: 

```python 

class Metrics:
    '''
    分类模型的评价指标。
    '''

    def __init__(self, y, y_pred, classes=None):
        '''
        初始化。y必须从0开始。

        Parameters
        ----------
        y : jnp.ndarray
            真实标签。
        y_pred : jnp.ndarray
            预测标签。
        classes : int
            分类的类别，通常在样本数量较小，或者无法包含所有类别的时候用，默认是None。
        '''

        self.y = y
        self.y_pred = y_pred

        if classes is not None:
            self.classes = classes
        else:
            uni = jnp.unique(self.y)
            self.classes = uni.shape[0]  # get classes num

        self.matrix = jnp.zeros((self.classes, self.classes))  # get confusion matrix

        if len(self.y.shape) == 1:
            temp_y = self.y
        elif len(self.y.shape) == 2:
            temp_y = jnp.argmax(self.y, axis=1)
        else:
            raise ValueError('Input y must be 1D or 2D.')

        if len(self.y_pred.shape) == 1:
            temp_y_pred = self.y_pred
        if len(self.y_pred.shape) == 2:
            temp_y_pred = jnp.argmax(self.y_pred, axis=1)
        elif len(self.y_pred.shape) > 2:
            raise ValueError('Input y_pred must be 1D or 2D.')

        for i, j in zip(temp_y, temp_y_pred):
            self.matrix = self.matrix.at[i, j].set(
                self.matrix[i, j] + 1
            )
```
to make code more rubust.

**Comment by HugoPhi** [comment]
Run Successfully, 
![Image](https://github.com/user-attachments/assets/43631f1f-6d28-4595-b248-66f324a83dfa)
